### DISCUSSION TITLE
How do I get started with iPhone development?

### DISCUSSION THREAD

**Discussion by Math-Grill** [community/community#21398] [Other Feature Feedback, Questions, & Ideas]
<p>iPhone, Swift, macOS Mojave </p>
<p>I am new to github and don’t know how to proceed ?</p>
<p>I’m reading the files and help files … I am lost in a wealth of data !!</p>
<p>I have a simple program I’d like to build in Swift and Xcode but am new to that as well !</p>
<p>I need help learning the basics of iPhone programming, from scratch :wink:</p>
<p>This Programming Help looks like a good starting place ?</p>
<p> - - - </p>
<p>I programmed a Math Grill some years  ago, It would present a math problem  multiplication and await an answer,  If the answer was wrong it would display the correct answer and present a new problem to the user, or just oresent a new problem.  No frills and stuff.  Just a workout of your math solving skills.  Training yourself to randomly recall the answers. !!!</p>
<p>This was in BASIC on an AMIGA 600 computer :wink:</p>
<p>I would like to build it on my iPhone 6+  I have an iMac OS Mojave and Xcode 10 and Swift 4.2.  but no experience in programming the Apple - iPhones or computers. </p>
<p>Yes I could use some help finding out how to get started ??</p>
<p>How to display text on the iPhone ??</p>
<p>How to enter text, numbers, into the program from a keypad display on the iPhone ??</p>
<p>How to get at the timer or time of day or a random number seed ??</p>
<p>And how to store and retrieve text from storage, a file, on the iPhone ??</p>
<p>Are all things of interest to me  ;) </p>
<p>Where do I find help to get started on my project ??</p>
<p>Thanks you for your help :))   Tammy</p>

**Comment by ThirdEyeRose** [comment, marked as answer]
<p>Hi <a class="mention" href="/math-grill">@math-grill</a>,</p>
<p>I moved your reply into its own post so that people are more aware of it and can respond to your questions directly. It’s helpful to create a new post for each new topic in general.</p>
<p>Regarding your questions, it seems like you’re biting off quite a lot at one time and might be a bit lost because of the sheer scope of your project. My recommendation would be to break your project down into much smaller pieces and tackle those separately.</p>
<p>If you’ve never created a program in Swift before, maybe a good start would be just to pick up a basic tutorial on Swift and work through it, taking note of areas where you have questions and where you get stuck. If you want to learn GitHub and Git first, you could have a look at the <a href="https://lab.github.com" rel="nofollow noopener">GitHub Learning Lab</a> and go through a couple of the courses just to get a feel for what Git and GitHub can do for your learning process.</p>
<p>I haven’t done any programming for iOS before, so I’m sorry that I don’t have more specific recommendations than that. Perhaps other people in the community have some recommendations on some good tutorials to get your toes wet in Swift. If you get stuck though, let us know where you get stuck and maybe we can help.</p>
<p>Thanks!</p>

**Comment by Math-Grill** [comment]
<p>Yes, Thank You  <span class="mention">@that_pat</span>  ;)</p>
<p>I’ll look into  <a href="https://lab.github.com/" rel="nofollow noopener">https://lab.github.com/</a>  it’s new to me, don’t tthink I’ve seen it :) </p>
<p>It will help me learn GitHub - I assume. ??</p>
<p>I’ve still got the programming problems I mentioned, that I need help with :( </p>
<p>Oh well, perhaps the Learning Lab will help ?</p>

**Comment by ThirdEyeRose** [comment]
<p>Hi <a class="mention" href="/math-grill">@math-grill</a>,</p>
<p>Yes, the <a href="https://lab.github.com" rel="nofollow noopener">GitHub Learning Lab</a> is for learning how to use Git and GitHub and some closely associated tools, that is correct. We don’t have any tutorials on iOS development, but you might find some through searching the web.</p>
<p>Thanks!</p>
